PlanetWorker/src/PlanetWorker.cs

```diff
@@ -56,19 +67,140 @@ static int Main(string[] arguments)
-          //     var pingResponse = new Pong(WorkerType, String.Format("{0}, World!", greeting));
-          //     var commandResponse = new PingResponder.Commands.Ping.Response(pingResponse);
-          //     connection.SendCommandResponse(request.RequestId, commandResponse);
-          // });
+          dispatcher.OnCommandRequest<AssignPlanetResponder.Commands.AssignPlanet>(request =>
+          {
+            var logMessage = String.Format("Received AssignPlanet command from player {0}", request.Request.Get().Value.playerId);
+            connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+              
+              foreach (KeyValuePair<EntityId, ViewEntity> pair in EntityView)
+              {
+                if(!pair.Value.hasAuthority)
+                {
+                  logMessage = String.Format("Skipping planet with entityId {0} because I don't have authority", pair.Key);
+                  connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+                  continue;
+                }
+                
+                logMessage = String.Format("Picked planet with entityId {0} because I have authority", pair.Key);
+                connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+                
+                logMessage = String.Format("Entity.PlanetInfo: {0}", pair.Value.entity.Get<PlanetInfo>());
+                connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+                
+                Option<IComponentData<PlanetInfo>> option;
+                IComponentData<PlanetInfo> planetInfo;
+                PlanetInfo.Data planetInfoData;
+                PlanetInfoData pid;
+
+                option = pair.Value.entity.Get<PlanetInfo>();
+                planetInfo = option.Value;
+                planetInfoData = planetInfo.Get();
+                pid = planetInfoData.Value;
+                
+                logMessage = String.Format("Entity {0} has playerId {1}", pair.Key, pid.playerId);
+                connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+                
+                if(pid.playerId == "")
+                {
+                  var planetId = pair.Key;
+                  var playerId = request.Request.Get().Value.playerId;
+                  
+                  //Create new component update object
+                  PlanetInfo.Update piu = new PlanetInfo.Update();
+                  piu.SetPlayerId(playerId);
+                  
+                  // Send the updates
+                  connection.SendComponentUpdate<PlanetInfo>(planetId, piu);
+                  
+                  // Send the assigned planet to the client
+                  var assignPlanetResponse = new AssignPlanetResponse(planetId);
+                  var commandResponse = new AssignPlanetResponder.Commands.AssignPlanet.Response(assignPlanetResponse);
+                  connection.SendCommandResponse(request.RequestId, commandResponse);
```

Comment:
IMO need to send a fail response if there were no planets available for this player - otherwise it will time out and that will be hard to debug (is it a real timeout or was there no planet?)